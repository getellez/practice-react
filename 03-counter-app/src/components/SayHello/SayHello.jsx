import React from 'react'
import PropTypes from 'prop-types';

export const SayHello = ({ title, subtitle, name }) => {
  return (
    <>
      <h1 data-testid="mytestid">Hola, { name }!</h1>
      <h2> { title } </h2>
      <small>{ subtitle }</small>
    </>
  )
}

SayHello.propTypes = {
  name: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
}

SayHello.defaultProps = {
  name: 'Germ√°n',
  title: 'Learning React',
  subtitle: 'The most popular frontend library',
}