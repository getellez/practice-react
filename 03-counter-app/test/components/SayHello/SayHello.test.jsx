import { render } from '@testing-library/react'
import { SayHello } from '../../../src/components/SayHello/SayHello.jsx'

describe('<SayHello> component', () => {
  
  it('should show the name in a H1', () => {
    const expectedTitle = 'Hola, Germán!';
    const name = 'Germán';
    const { container, getByTestId } = render(<SayHello name={name} />)
    expect(getByTestId('mytestid').innerHTML).toContain(expectedTitle)
  });
  it('should show the title in a H2', () => {
    const title = 'Learning React';
    const { container } = render(<SayHello title={title}/>)
    const h2 = container.querySelector('h2')
    expect(h2.innerHTML).toContain(title)
  });

  it('debe de hacer match con el snapshot', () => {
    const name = 'Germán'
    const { container } = render(<SayHello name={name} />)
    expect(container).toMatchSnapshot()
  })

})