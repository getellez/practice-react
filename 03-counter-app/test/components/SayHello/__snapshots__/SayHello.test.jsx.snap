// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SayHello> component debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="mytestid"
  >
    Hola, 
    Germ√°n
    !
  </h1>
  <h2>
     
    Learning React
     
  </h2>
  <small>
    The most popular frontend library
  </small>
</div>
`;
